pandas/core/frame.py

```diff
@@ -8195,6 +8195,10 @@ def update(
         for col in self.columns:
             this = self[col]._values
             that = other[col]._values
+
+            if all(isna(that)):
```

Comment:
From the original issue, it suggests that this may be an issue in the `reindex_like` operations being incompatible with something, so I don't think a patched-over check like this is appropriate

Reply:
Hi! Thanks for reviewing! 

This was my assessment of it in the original issue: https://github.com/pandas-dev/pandas/issues/16713#issuecomment-1296122554

Based on that, I think the issue in reindex_like is just that when it creates columns that aren't in the pre-reindex dataframe that they aren't datetime/datetime compatible by default. Is that the part I should be looking to change instead?

Reply:
> Is that the part I should be looking to change instead?

Ideally yes, or possibly when the masks are compared.

Reply:
Will look into this more, thanks for the guidance!